src/sonic-eventd/src/loghandler.cpp

```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+#include <syslog.h>
+#include <cstring>
+
+
+
+// Safe fallback for null or empty C-style strings
+const char* safe(const char* s) {
+    return (s && std::strlen(s) > 0) ? s : "";
+}
+
+
+extern "C" void openSyslog() {
+    openlog (NULL, LOG_CONS | LOG_PID | LOG_NDELAY, LOG_LOCAL4);
+}
+
+extern "C" void writeToSyslog(char* ev_id, int ev_sev, char* ev_type, char* ev_act,
```

Comment:
.h definition takes in const char*, shouldnt this follow?

Reply:
Fixed.